editor/code_editor.cpp

```diff
@@ -419,6 +419,26 @@ void FindReplaceBar::_update_matches_label() {
 	}
 }
 
+void FindReplaceBar::_temp_unfold_line(int p_line) {
+	// Unfold p_line, then walk back to find which was the first line that had to be unfolded so p_line can be shown.
+	// Mark that line as temporarily unfolded, so it's folded back on the next temporarily unfolding - unless the user for example edits the text.
+	if (temp_unfolded_line != -1) {
+		text_editor->fold_line(temp_unfolded_line);
+	}
+
+	if (!text_editor->is_line_folded(p_line)) {
+		temp_unfolded_line = p_line;
+		while (temp_unfolded_line > 0) {
```

Comment:
This will always result in `temp_unfolded_line` being `>= 0`, is there any guarantee that there will be a folded line on the way here, or will this always fold line 0 if it is unfolded? As it won't ever reach -1

Reply:
It does indeed have that bug, will provide a suggestion